### PR TITLE
Fixed go-agents dependencies

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Fixes go-agents dependencies:
1. Previously we have `major range` dependencies declaration which is potentially unreliable. So, dependencies are declared as `equals` versions.
2. `che-go-jsonrpc` was missing in `Gopkg.tomm`. So, added it.

### What issues does this PR fix or reference?
This issue is discovered while making changes on `go-agent` side for self-signed-certificates https://github.com/eclipse/che/pull/12089

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

**Comment by sleshchenko** [comment]
ci-test

**Review by skabashnyuk — APPROVED** [review]

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1267/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1267//Selenium_20tests_20report/)
docker image: eclipseche/che-server:12111
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

File: agents/go-agents/Gopkg.lock [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 
 [[projects]]
-  digest = "1:76dc72490af7174349349838f2fe118996381b31ea83243812a97e5a0fd5ed55"
+  digest = "1:217f778e19b8d206112c21d21a7cc72ca3cb493b67631680a2324bc50335d432"
   name = "github.com/dgrijalva/jwt-go"
   packages = ["."]
   pruneopts = "UT"
-  revision = "06ea1031745cb8b3dab3f6a236daf2b0aa468b7e"
-  version = "v3.2.0"
```
Comment (garagatyi) [inline agents/go-agents/Gopkg.lock:10]: Do we have CQ for this version?

Comment (skabashnyuk) [reply agents/go-agents/Gopkg.lock:10]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15288

**Pull request merged**